app/js/components/manageApps/AddAddon.jsx

```diff
@@ -6,68 +6,64 @@
  * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
  * graphic logo is a trademark of OpenMRS Inc.
  */
-import React from 'react';
+import React, { Component } from 'react';
 import PropTypes from 'prop-types';
+import Dropzone from 'react-dropzone';
 
-export default function AddAddon({
-  handleClear, 
-  handleUpload, 
-  displayManageOwaButtons, 
-  displayManageOwaButtonsState }) {
-  const classesToUse = displayManageOwaButtonsState ?
-    'col-xs-6 col-sm-8' : 'col-xs-6 col-sm-12';
-  return (
-    <div className="container-fluid">
-      <fieldset className="scheduler-border">
-        <legend className="scheduler-border">Upload Addon Package:</legend>
-        <div className="row">
-          <div className="col-xs-12">
-            <div className={classesToUse}>
-              <input
-                name="file"
-                type="file" 
-                id="fileInput"
-                onChange={displayManageOwaButtons}
-                accept=".zip"
-                data-text="Browse Addon"
-                data-buttonName="btn-primary"
-                data-placeholder="Select Addon to upload"
-                data-btnClass="btn-primary"
-                className="filestyle" />
-            </div>
-            {
-              displayManageOwaButtonsState && 
-                <div className="col-xs-6 col-sm-4">
-                  <div className="row">
-                    <div className="btn-toolbar">
-                      <button
-                        className="btn btn-primary"
-                        onClick={handleUpload}>
-                        <span
-                          className="glyphicon glyphicon-upload"
-                        /> Upload
-                      </button>
-                      <button
-                        id="cancel-btn"
-                        className="btn btn-danger"
-                        onClick={handleClear}
-                      >
-                        <span className="glyphicon glyphicon-remove" /> Clear
-                      </button>
-                    </div>
-                  </div>
-                </div>
-            }
-          </div>
+export default class AddAddon extends Component {
+  constructor() {
+    super();
+    this.state = {
+      disableClick: true,
+      dropzoneRef: {},
+    }
+    this.showMessage = this.showMessage.bind(this);
+  }
+
+  showMessage() {
+    if (this.props.files == null || this.props.files.length < 1) {
```

Comment:
`===` instead of `==`

Reply:
The alternative is that a file is set i.e. props has a Filelist object. The initial value of files is null, set in ManageApps state and passed on to AddAddon via props. When a zip file is dragged and dropped, the files state is set to the Filelist object. When a non-zip file is drag and dropped the files state is set to an empty list